service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -96,6 +96,11 @@ public String mapObservationToObservationStatement(Observation observation, bool
     private String prepareComment(Observation observation) {
         StringBuilder commentBuilder = new StringBuilder(observation.hasComment() ? observation.getComment() : StringUtils.EMPTY);
 
+        if (observation.hasComponent()) {
+            commentBuilder.insert(COMMENT_OFFSET,
```

Comment:
Optional improvement, but boy scout rule :) Instead of using StringBuilder::insert can we just append `observation.hasComment() ? observation.getComment() : StringUtils.EMPTY` as the very last step of this method?

Have a look at what insert does. Every time we call it the entire string is shifted (copied) to the right by N places. This will not be very performant.